go/iot/arduino/protocol.txt

```diff
@@ -36,23 +36,28 @@ When the program starts, the following view would be registered.
 
 Before the registration, the Raspberry Pi would not make any response to the the request from Arduino.
 
-After the registration, the Raspberry Pi would make a negative or positive response to any record request from Arduino.
-
-When the Pi would make a negative response:
-    1. The measure doesn't bind to any views.
-    2. The tag key doesn't bind to any views.
-    2. Pi could not parse the encapsulated data because of the format error.
-    3. The number of tag keys is not equal to the number of tag values.
-
-When the Pi successfully parses and records data, it would make a positive response to the request. Note that it would
-still responds positively if there is something wrong with the OpenCensus framework such as networking failure.
+After the registration, the Raspberry Pi would make a response to the record request from Arduino.
 
 Typical example for the response by Raspberry Pi would be as below:
 {
 	"Code": 200 or 404
 	"Info": "Record Successfully!" or "Registration Not Finished" or ...
 }
 
+In the following cases Pi would make response with the Code 404:
+    1. The measure doesn't bind to any views.
```

Comment:
Sorry I missed this one - I don't think we should return 404 in this case. If there are no views associated with this measure, recording will just be a no-op, but it shouldn't be an error. Consider logging the corresponding measure name on Pi instead.

See https://github.com/census-instrumentation/opencensus-go/blob/212f272cd24d9e931330f7e68a48357050fd3c18/stats/view/worker_commands.go#L149-L152.

Reply:
Instead, if the OpenCensus library returns any error when trying to record the measurements, you can send a 404 back to Arduinos.

Reply:
It sounds strange that I return a 200 when the measure is not registered before.  Can I add a new code for this kind of situation?  Otherwise the end of arduino might never know something wrong happens.

Reply:
Sounds good, you can send another code to notify Arduino that measure is not registered and do not send request any more. (There's no meaning to retry in this case.)